msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/AcquireTokenSilentlyTest.java

```diff
@@ -116,6 +119,79 @@ void confidentialAppAcquireTokenSilently_claimsSkipCache() throws Throwable {
         assertInstanceOf(MsalInteractionRequiredException.class, ex.getCause());
     }
 
+    @Test
+    void testTokenRefreshReasons() throws Exception {
+        DefaultHttpClient httpClientMock = mock(DefaultHttpClient.class);
+
+        ConfidentialClientApplication cca =
+                ConfidentialClientApplication.builder("clientId", ClientCredentialFactory.createFromSecret("password"))
+                        .authority("https://login.microsoftonline.com/tenant/")
+                        .instanceDiscovery(false)
+                        .validateAuthority(false)
+                        .httpClient(httpClientMock)
+                        .build();
+
+        HashMap<String, String> responseParameters = new HashMap<>();
+
+        //Acquire a token that expired an hour ago
+        responseParameters.put("access_token", "expiredToken");
+        responseParameters.put("id_token", TestHelper.createIdToken(new HashMap<>()));
+        responseParameters.put("expires_in", "-3600");
```

Comment:
I'd argue that MSAL should never accept to put this in the cache.

Reply:
Writing that test case definitely felt wrong, and I agree the library shouldn't allow it. For the most part we just extract various values from the token response, and don't check whether or not the values make sense: https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/dev/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenRequestExecutor.java

I saw that MSAL .NET does do some checks and other MSALs probably do some as well, so I'll create an SDK consistency issue to add in some validation.